src/test/java/uk/gov/hmcts/ethos/replacement/docmosis/reports/respondentsreport/RespondentsReportTest.java

```diff
@@ -0,0 +1,113 @@
+package uk.gov.hmcts.ethos.replacement.docmosis.reports.respondentsreport;
+
+import java.time.Clock;
+import java.time.Instant;
+import java.time.LocalDateTime;
+import java.time.ZoneId;
+import java.util.ArrayList;
+import java.util.List;
+import static org.junit.Assert.*;
+import static org.junit.Assert.assertEquals;
+import org.junit.Before;
+import org.junit.Test;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+import static uk.gov.hmcts.ecm.common.model.helper.Constants.*;
+import uk.gov.hmcts.ecm.common.model.reports.respondentsreport.RespondentsReportSubmitEvent;
+
+public class RespondentsReportTest {
+
+    RespondentsReportDataSource reportDataSource;
+    RespondentsReport respondentsReport;
+    RespondentsReportCaseDataBuilder caseDataBuilder;
+    List<RespondentsReportSubmitEvent> submitEvents = new ArrayList<>();
+    static final LocalDateTime BASE_DATE = LocalDateTime.of(2022, 1, 1, 0, 0,0);
+    static final String DATE_FROM = BASE_DATE.minusDays(1).format(OLD_DATE_TIME_PATTERN);
+    static final String DATE_TO = BASE_DATE.plusDays(15).format(OLD_DATE_TIME_PATTERN);
+    public RespondentsReportTest() {
+        caseDataBuilder = new RespondentsReportCaseDataBuilder();
+    }
+
+    @Before
+    public void setup() {
+        submitEvents.clear();
+        reportDataSource = mock(RespondentsReportDataSource.class);
+        when(reportDataSource.getData(MANCHESTER_CASE_TYPE_ID, DATE_FROM, DATE_TO)).thenReturn(submitEvents);
+        respondentsReport = new RespondentsReport(reportDataSource,DATE_FROM, DATE_TO);
+    }
+
+    @Test
+    public void shouldNotShowCaseWithNoRespondent() {
```

Comment:
Formatting
Rather than doing submits.clear and then caseDataBuilder = new RespondentsReportCaseDataBuilder();
You could just put these in the setup method
```
submitEvents.clear();
caseDataBuilder = new RespondentsReportCaseDataBuilder()
```

Reply:
refactored